django/db/models/expressions.py

```diff
@@ -1068,12 +1075,11 @@ def as_sql(self, compiler, connection, template=None, **extra_context):
             sql = 'NOT {}'.format(sql)
         return sql, params
 
-    def as_oracle(self, compiler, connection, template=None, **extra_context):
+    def select_format(self, compiler, sql, params):
         # Oracle doesn't allow EXISTS() in the SELECT list, so wrap it with a
-        # CASE WHEN expression. Change the template since the When expression
-        # requires a left hand side (column) to compare against.
-        sql, params = self.as_sql(compiler, connection, template, **extra_context)
-        sql = 'CASE WHEN {} THEN 1 ELSE 0 END'.format(sql)
+        # CASE WHEN expression.
+        if compiler.connection.vendor == 'oracle':
```

Comment:
Maybe we could add and use here a db feature e.g. `supports_boolean_in_select_clause` (we have e.g. `supports_default_in_lead_lag` so it seems reasonable).

Reply:
@felixxm do you know if _other_ boolean expressions work in `SELECT` for Oracle?

e.g. `(TABLE.COLUMN > 3) AS gt_3`?

If that's not the case then I guess this boolean flag could be used to add support for `.annotate(gt_3=Q(column__gt3))` in the future as well.

Reply:
No, they are not supported, because `BOOLEAN` datatype is available only in PL/SQL on Oracle, so `SELECT` clause cannot return it.

Reply:
@felixxm thanks for confirming.

Then a `supports_boolean_expr_in_select_clause` flag might be more appropriate then because `supports_boolean_in_select_clause` might be confusing for `Col(output_field=BooleanField())`?

Reply:
Agreed :+1: